File: peek_plugin_diagram/_private/both-app/tuples/LocationIndexUpdateDateTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {diagramTuplePrefix} from "@peek/peek_plugin_diagram/_private";


@addTupleType
export class LocationIndexUpdateDateTuple extends Tuple {
    public static readonly tupleName = diagramTuplePrefix + "LocationIndexUpdateDateTuple";

    modelSetKey:string;
    indexBucketUpdateDates: {};

    constructor() {
        super(LocationIndexUpdateDateTuple.tupleName)
    }
}

File: peek_plugin_diagram/_private/tuples/LocationIndexUpdateDateTuple.py
from typing import Dict

from peek_plugin_diagram._private.PluginNames import diagramTuplePrefix
from vortex.Tuple import addTupleType, TupleField, Tuple



#: This the type of the data that we get when the clients observe new locationIndexs.
DeviceLocationIndexT = Dict[str, str]

@addTupleType
class LocationIndexUpdateDateTuple(Tuple):
    __tupleType__ = diagramTuplePrefix + "LocationIndexUpdateDateTuple"

    modelSetKey: str = TupleField()
    indexBucketUpdateDates: DeviceLocationIndexT = TupleField({})
